components/net_traits/image_cache_thread.rs

```diff
@@ -13,27 +14,42 @@ use std::sync::Arc;
 /// and/or repaint.
 #[derive(Clone, Deserialize, Serialize)]
 pub struct ImageResponder {
-    sender: IpcSender<ImageResponse>,
+    id: PendingImageId,
+    sender: IpcSender<PendingImageResponse>,
+}
+
+#[derive(Deserialize, Serialize)]
+pub struct PendingImageResponse {
+    pub response: ImageResponse,
+    pub id: PendingImageId,
 }
 
 impl ImageResponder {
-    pub fn new(sender: IpcSender<ImageResponse>) -> ImageResponder {
+    pub fn new(sender: IpcSender<PendingImageResponse>, id: PendingImageId) -> ImageResponder {
         ImageResponder {
             sender: sender,
+            id: id,
         }
     }
 
     pub fn respond(&self, response: ImageResponse) {
-        self.sender.send(response).unwrap()
+        let _ = self.sender.send(PendingImageResponse {
```

Comment:
Since the semantics have changed here when the send fails, perhaps add a comment to explain when / why this happens (if it's expected), or add a warn!() or similar if it should never fail?